harness/determined/core/_context.py

```diff
@@ -100,6 +100,11 @@ def _dummy_init(
     distributed = distributed or core.DummyDistributedContext()
     preempt = core.DummyPreemptContext(distributed, preempt_mode)
 
+    if storage_manager is not None:
+        logging.warning("storage_manager is deprecated. Consider using checkpoint_storage instead.")
```

Comment:
There's a more pythonic way to emit a deprecation warning.  We have a few examples, try `git grep FutureWarning`.